enginetest/join_op_tests.go

```diff
@@ -106,6 +106,102 @@ var joinCostTests = []struct {
 			setup.XyData[0],
 		},
 		tests: []JoinOpTests{
+			{
+				Query: "select * from mytable where exists (select * from mytable where i = 1);",
+				Expected: []sql.Row{
+					{1, "first row"},
+					{2, "second row"},
+					{3, "third row"},
+				},
+			},
+			// queries that test subquery hoisting
+			{
+				// case 1: condition uses columns from both sides
+				Query: "/*case1*/ select * from ab where exists (select * from xy where ab.a = xy.x + 3)",
+				Expected: []sql.Row{
+					{3, 1},
+				},
+			},
+			{
+				// case 1N: NOT EXISTS condition uses columns from both sides
+				Query: "/*case1N*/ select * from ab where not exists (select * from xy where ab.a = xy.x + 3)",
+				Expected: []sql.Row{
+					{0, 2},
+					{1, 2},
+					{2, 2},
+				},
+			},
+			{
+				// case 2: condition uses columns from left side only
+				Query:    "/*case2*/ select * from ab where exists (select * from xy where a = 1)",
+				Expected: []sql.Row{{1, 2}},
+			},
+			{
+				// case 2N: NOT EXISTS condition uses columns from left side only
+				Query: "/*case2N*/ select * from ab where not exists (select * from xy where a = 1)",
+				Expected: []sql.Row{
+					{0, 2},
+					{2, 2},
+					{3, 1},
+				},
+			},
+			{
+				// case 3: condition uses columns from right side only
+				Query: "/*case3*/ select * from ab where exists (select * from xy where 1 = xy.x)",
+				Expected: []sql.Row{
+					{0, 2},
+					{1, 2},
+					{2, 2},
+					{3, 1},
+				},
+			},
+			{
+				// case 3N: NOT EXISTS condition uses columns from right side only
+				Query: "/*case3N*/ select * from ab where not exists (select * from xy where 10 = xy.x)",
+				Expected: []sql.Row{
+					{0, 2},
+					{1, 2},
+					{2, 2},
+					{3, 1},
+				},
+			},
+			{
+				// case 4a: condition uses no columns from either side, and condition is true
+				Query: "/*case4a*/ select * from ab where exists (select * from xy where 1 = 1)",
+				Expected: []sql.Row{
+					{0, 2},
+					{1, 2},
+					{2, 2},
+					{3, 1},
+				},
+			},
+			{
+				// case 4aN: NOT EXISTS condition uses no columns from either side, and condition is true
+				Query:    "/*case4aN*/ select * from ab where not exists (select * from xy where 1 = 1)",
+				Expected: []sql.Row{},
+			},
+			// TODO: uncomment after https://github.com/dolthub/dolt/issues/5522 is fixed
```

Comment:
Do we still need these TODOs? Looks like that issue is fixed and we already uncommented the tests?